src/components/Table/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+const Wrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.surface3};
+  border-radius: 20px;
+  overflow: hidden;
+`
+const TableContainer = styled.div`
+  overflow: auto; // Enable horizontal scroll on full table
+`
+const TableHeadContainer = styled.div`
+  // Enable horizontal scroll on the header but hide scrollbar
+  overflow-x: auto;
+  ::-webkit-scrollbar {
+    display: none;
+  }
+  -ms-overflow-style: none;
+  scrollbar-width: none;
```

Comment:
adding `overscroll-behavior: none;` will stop the bouncing on the header when overscrolled